inc/msp/byte_vector.hpp

```diff
@@ -0,0 +1,263 @@
+#ifndef BYTE_VECTOR_HPP
+#define BYTE_VECTOR_HPP
+
+#include "value.hpp"
+
+#include <vector>
+#include <limits>
+#include <type_traits>
+#include <memory>
+#include <cstdint>
+
+namespace msp {
+
+
+struct Packable;
+/**
+ * @brief ByteVector vector of bytes
+ */
+
+class ByteVector : public std::vector<uint8_t>
```

Comment:
Add doxygen documentation about what classes and methods do.
In QtCreator you can type `/**` + Enter to create a template with the correct method names, parameters, and outputs.